python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,22 +365,24 @@ def __get__(self, obj, cls):
             return cls._schema[name]
 
         if self._getter_name:
-            return self._bind_method(obj, self._getter_name)()
+            returnable = self._bind_method(obj, self._getter_name)()
+            if isinstance(returnable, GangaObject):
+                returnable._setParent(self)
+            return returnable
```

Comment:
I feel it would be technically cleaner if we just made sure that the parent was being set correctly everywhere but I see your point that the getter can be a special case. There's no need to revert and I will merge this soon.